packages/dds/tree/tsconfig.json

```diff
@@ -14,6 +14,7 @@
 		// Linter is used to enforce "_" prefix for unused locals to prevent accidentally having unused locals.
 		"noUnusedLocals": false,
 		"noImplicitOverride": true,
+		"lib": ["ES2020", "ES2022.Array", "DOM", "DOM.Iterable"],
```

Comment:
I don't recall exactly what the effect of this is. Makes APIs available for us to use in source, but they still get transpiled into whatever target we tell TSC to use?

Reply:
lib provides type declarations for built-ins.
We need this because `tree` uses `array.at`. Before, node types were bleeding in. This has no impact on result code as is mostly the case with typescript.
I did not look into the `array.at` introduction to see if it might be a problem. Was deferring to its introduction as confirming okay to use. @Abe27342, any need to follow-up per use added in #18689?

Reply:
Wow, I didn't realize `.at` was a 2022 feature. Nobody's complained about it yet, but tree hasn't seen much widespread usage and in other places we explicitly avoid going that new (I've even seen es2021 features shy-ed away from in use). Opened a PR to remove it: #19904.

If your PR here merges first, I'll update that to remove ES2022.Array here.